### PR TITLE
#2163: Update mapstore to latest 2022.01.xx

### PR DESCRIPTION
**Pull request by dsuren1**
## Description
- The Mapstore submodule has been updated to latest stable of 2022.01.xx
- The application specific config are retained and adjusted wherever applicable

## Issue
[#2163](https://github.com/geosolutions-it/support/issues/2163)

### CODE REVIEW

**Review by MV88 — CHANGES_REQUESTED** [review]
@dsuren1 please check my comments and let me know if you need clarifications

- [ ] npm install does not work cloning from scratch the project because it is missing .npmrc in root directory, please add it
- [ ] please update all imports in js folder to use aliases @js @mapstore
- [ ] extension installation seem to not work
- [ ] search by bookmark is not present
- [ ] DashboardEditor configuration is not working
image

File: configs/localConfig.json [hunk 1]
```diff
@@ -914,6 +902,32 @@
             }
           }
         ],
+        "geostory-embedded": [
```
[the remark this answers is not in the document]
Comment (dsuren1) [reply configs/localConfig.json:918]: Updated missing ones in latest commit

File: configs/localConfig.json [hunk 2]
```diff
@@ -489,19 +489,6 @@
                 "cfg": {
                     "wrap": true
                 }
-            }, {
-                "name": "Identify",
```
[the remark this answers is not in the document]
Comment (dsuren1) [reply configs/localConfig.json:493]: There seems to have been two instance of `Identify` plugin in desktop mode

File: index.html [hunk 3]
```diff
@@ -3,7 +3,7 @@
     <head>
         <meta charset="UTF-8">
         <meta http-equiv="Content-Type" content="text/html;charset=UTF-8">
-        <meta name="viewport" content="width=device-width, initial-scale=1">
+        <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1.0, user-scalable=no">
```
[the remark this answers is not in the document]
Comment (dsuren1) [reply index.html:6]: Updated indexTemplate

File: web/src/main/webapp/WEB-INF/extensions-servlet.xml [hunk 4]
```diff
@@ -0,0 +1,22 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<beans xmlns="http://www.springframework.org/schema/beans"
+    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+    xmlns:p="http://www.springframework.org/schema/p"
```
[the remark this answers is not in the document]
Comment (dsuren1) [reply web/src/main/webapp/WEB-INF/extensions-servlet.xml:4]: I have tested it from context. Works fine now

File: web/src/main/resources/applicationContext.xml [hunk 5]
```diff
@@ -29,76 +26,32 @@
...
-        </property>
-        <!-- Formats supported by the converter -->
-        <property name="formats">
+    <!-- Allows getting database configuration also from the data-dir -->
+    <bean class="org.springframework.beans.factory.config.PropertyOverrideConfigurer">
+        <property name="ignoreResourceNotFound" value="true"/>
+        <property name="order" value="10"/>
+        <property name="locations">
             <list>
-                <value>jpg</value>
```
[the remark this answers is not in the document]
Comment (dsuren1) [reply web/src/main/resources/applicationContext.xml:84]: Context for print is loaded from `src\printing\applicationContext-print.xml`, all previous and new configs are applied properly

**Comment by dsuren1** [comment]
> @dsuren1 please check my comments and let me know if you need clarifications
> 
> * [ ]  npm install does not work cloning from scratch the project because it is missing .npmrc in root directory, please add it

I have added an empty .npmrc file now

> * [ ]  search by bookmark is not present

The search by bookmark is enabled only for admin user. And works fine.

File: webpack.config.js [hunk 6]
```diff
@@ -29,20 +29,30 @@ module.exports = require('./MapStore2/build/buildConfig')(
     },
     {
         '/rest/geostore': {
-            target: "https://austrocontrol.geo-solutions.it/austrocontrol-ms2",
+            target: "http://localhost:8080/mapstore",
```
Comment (MV88) [inline webpack.config.js:32]: do we want this to be published?
maybe is fine to point to the this instance while developing, please leave it as a comment or restore it

**Review by MV88 — CHANGES_REQUESTED** [review]
all good just a couple of things

File: webpack.config.js (hunk 6 above)
Comment (dsuren1) [reply webpack.config.js:32]: I shall restore it to `https://austrocontrol.geo-solutions.it/austrocontrol-ms2` for all proxy path. Maybe the client had this for their convenience. However until the new changes are deployed we may have to reply on local instance to test feature like 'extension'.

Comment (MV88) [reply webpack.config.js]: What about having a dedicated webpack-config for local testing, where we override only the proxy?

Comment (MV88) [reply webpack.config.js]: this would require a change in npm start command

Comment (MV88) [reply webpack.config.js]: or adding a specific command like "npm startDev run" that points to austrocontrol.geo-solutions.it etc

**Review by MV88 — APPROVED** [review]

File: webpack.config.js (hunk 6 above)
Comment (MV88) [reply webpack.config.js]: thinking twice, let's leave it as it is now, with reference to their instance

**Comment by MV88** [comment]
@ElenaGallo testable in DEV

**Pull request merged**